src/lib.rs

```diff
@@ -96,15 +98,22 @@ pub mod prelude {
 /// ```
 pub fn spawn<T: Send + 'static>(future: impl Future<Output = T> + Send + 'static) -> Task<T> {
     static GLOBAL: Lazy<Executor> = Lazy::new(|| {
-        thread::Builder::new()
-            .name("smol".to_string())
-            .spawn(|| {
-                loop {
+        let num_threads = {
+            // Parse SMOL_THREADS or use the number of CPU cores on the system.
+            std::env::var("SMOL_THREADS")
+                .ok()
+                .and_then(|s| s.parse().ok())
+                .unwrap_or(1)
+        };
+        for n in 0..num_threads {
+            thread::Builder::new()
+                .name(format!("smol-{}", n))
```

Comment:
I'd prefer numbering threads from 1 to n rather than 0 to n-1. That's how threads are numbered in the `blocking` crate and (I believe) is generally a more common pattern.